lib/cartopy/mpl/feature_artist.py

```diff
@@ -189,8 +219,7 @@ def draw(self, renderer, *args, **kwargs):
                     projected_geom = geom
 
                 geom_paths = cpatch.geos_to_path(projected_geom)
-                if not geom_paths:
-                    continue
```

Comment:
If we call `make_compound_path` with zero paths then it [returns an empty path](https://github.com/matplotlib/matplotlib/blob/810a43b3382125da7406c548f837b1d1b4b3833b/lib/matplotlib/path.py#L324-L325).  Keeping hold of this helps us preserve one-path-per-feature.